tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+  output-type: lib | exe     # generate executable (default) or library
+  processor:                 # specify processor attributes 
+  groups:                    # source files organized in groups
+  layers:                    # software layers of pre-configured software components
+  components:                # software components
+```  
+
+
+## Processor Attributes
+
+The `processor:` keyword defines attributes such as TrustZone and FPU register usage.
+
+```yml
+ processor:                # processor specific settings
+    trustzone: secure      # TrustZone mode: secure | non-secure | off
+    fpu: on | off          # control usage of FPU registers (S-Register for Helium/FPU) (default: on for devices with FPU registers)
+    endian: little | big   # select endianess (only available when devices are configureable)
+```
+
+**Note:**
+- Default for `trustzone:` 
+   - `off` for devices that support this option.
+   - `non-secure` for devices that have TrustZone enabled.
+   
+
+# Pre/Post build steps
+
+Tbd: potentially map to CMake add_custom_command.
+
+```yml
+- execute: description      # execute an external command with description
+  os: Linux                 # executed on which operating systems (if omitted it is OS independent)
+  run:                      # tool name that should be executed, optionally with path to the tool
+  args:                     # tool arguments
+  stop:                     # stop on exit code
+```
+
+Potential usage before/after build
+
+```yml
+solution:
+  - execute: Generate Keys for TF-M
+    os: Linux
+    run: KeyGen.exe
+  - project: /security/TFM.cproject.yml
+  - project: /application/MQTT_AWS.cproject.yml
+  - execute: Copy output files
+    run: cp *.out .\output
+```
+
+Potential usage during build steps
+```yml
+groups:
+  - group:  "Main File Group"
+    files: 
+      - execute: Generate file1a.c
+        run: xyz.exe
+        ....
+      - file: file1a.c
+```
+
+# Layer
+
+Start of a layer definition in a `*.clayer.yml` file.
+
+
+**Example:**
+
+
+
+```yml
+project:
+  compiler: AC6
+
+  layers:
+    - layer: ..\layer\App\Blinky.clayer.yml
+    - layer: ..\layer\Board+NUCELO-G474RE\Nucelo-G474RE.clayer.yml      # 'NUCLEO-G474RE' is target-type
+    - layer: ..\layer\RTOS\RTX.clayer.yml
+```    
+
+
+
+```yml
+layer: App
+# name: Blinky
+#  description: Simple example
+
+# interfaces:
+#   consumes:
+#     - C_VIO:
+#     - RTOS2:
+
+  groups:
+    - group: App
+      files:
+        - file: "./Blinky.c"
+    - group: Documentation
+      files:
+        - file: name="./README.md"
+```
+
+```yml
+layer:
+  type: RTOS
+  description: Keil RTX5 open-source real-time operating system with CMSIS-RTOS v2 API
+  
+interface:
+  provides:
+    - RTOS2:
+    
+components:
+  component: CMSIS:RTOS2:Keil RTX5&Source
+# open how do we capture versions of config files?
+```
+
+```yml
+layer: Board
+  name: NUCLEO-G474RE
+  description: Board setup with interfaces 
+  
+interfaces:
+  consumes:
```

Comment:
consumes / provides  concepts not clear at this stage

Reply:
agreed needs more work